r/R/metadata.R

```diff
@@ -115,7 +115,7 @@ apply_arrow_r_metadata <- function(x, r_metadata) {
 remove_attributes <- function(x) {
   removed_attributes <- character()
   if (identical(class(x), c("tbl_df", "tbl", "data.frame"))) {
-    removed_attributes <- c("class", "row.names", "names")
+    removed_attributes <- c("class", "row.names", "names", "groups")
```

Comment:
It might be hard to find a definitive answer if this is ok, but this will remove any `groups` attribute whether or not it is from dplyr grouping. It is almost certainly ok, since dplyr _itself_ would overwrite this attribute if something used it, but in this code path, we aren't any more checking `inherits(x, "grouped_df")` before deleting that attribute.

Thoughts?

Reply:
Good catch! Have updated the PR to account for this - let me know if it looks alright to you.